src/JSON/Utils/Views.dfy

```diff
@@ -0,0 +1,121 @@
+include "../../BoundedInts.dfy"
+
+module {:options "-functionSyntax:4"} JSON.Utils.Views.Core {
+  import opened BoundedInts
+
+  type View = v: View_ | v.Valid? witness View([], 0, 0)
+  datatype View_ = View(s: bytes, beg: uint32, end: uint32) {
```

Comment:
These fields (and the constructor) are intended to be internal, so keeping them short and (somewhat) opaque isn't necessarily a bad thing, I think.  It's important for encapsulation to not peek at the underlying string or at the bounds.

In contrast, renaming `s` to `bytes` would be misleading, because there is a public function `Bytes` that gives you `s[beg..end]`.

That said, I'm OK to rename if you're not convinced.